### PR TITLE
Remove usage of Contexts from grpc instrumentation.

### PR DESCRIPTION
**Pull request by anuraaga**
And add a smoke test.

I originally was just trying to add a smoke test and found our agent is broken :) `Contexts.interceptCall` seems to be hard for us to use in instrumentation since `Contexts` needs to be unshaded, but `Context` is shaded. This problem will go away in #1384 when we don't use gRPC context.

As a quick fix, I have restored manual propagation of span - we should be able to cherrypick this into the release branch if we want to fix grpc support in the agent. Note, I am propagating span, not context, which we don't like to do, but I think clobbering user context is worse than losing baggage or whatever may be lost - and it's just temporary.

### CODE REVIEW

**Comment by anuraaga** [comment]
I wasn't able to get the smoke test to pick up a new version of the javaagent without disabling gradle cache to rebuild it - not sure but a bit worried the build failure is the same cache issue.

File: .github/workflows/build-grpc-smoke-dist.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Build Spring Boot smoke test distribution
```
Comment (iNikem) [inline .github/workflows/build-grpc-smoke-dist.yaml:1]: That's not Spring Boot, right?

File: .gitignore [hunk 2]
```diff
@@ -58,7 +57,4 @@ hs_err_pid*
 replay_pid*
 
 !java-agent/benchmark/releases/*.jar
-/smoke-tests/play/.gradle/
```
Comment (iNikem) [inline .gitignore:61]: Why?

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/GrpcSmokeTest.groovy [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    [currentAgentVersion] as Set == findResourceAttribute(traces, "telemetry.auto.version")
+      .map { it.stringValue }
+      .collect(toSet())
+
+    cleanup:
+    stopTarget()
+
+    where:
+    jdk << [11]
+    // jdk << [8, 11, 15]
```
Comment (iNikem) [inline smoke-tests/src/test/groovy/io/opentelemetry/smoketest/GrpcSmokeTest.groovy:52]: Wanna change back?

File: smoke-tests/grpc/build.gradle [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  implementation "io.grpc:grpc-stub:1.32.1"
+  implementation "io.opentelemetry:opentelemetry-proto:0.9.1"
+  implementation "io.opentelemetry:opentelemetry-extension-auto-annotations:0.9.1"
+  implementation "org.apache.logging.log4j:log4j-core"
+
+  runtimeOnly("org.apache.logging.log4j:log4j-slf4j-impl")
+}
+
+compileJava {
+  options.release = 11
```
Comment (mateuszrzeszutek) [inline smoke-tests/grpc/build.gradle:27]: Shouldn't it be 8 here? If you want it to run on Java 8

File: .gitignore (hunk 2 above)
Comment (anuraaga) [reply .gitignore:61]: I made .gradle a non-absolute path above which seems like low chance of collision :)

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/GrpcSmokeTest.groovy (hunk 3 above)
Comment (anuraaga) [reply smoke-tests/src/test/groovy/io/opentelemetry/smoketest/GrpcSmokeTest.groovy:52]: Doh yeah

**Review by trask — APPROVED** [review]
🎉 new smoke test 🎉

**Pull request merged**